src/Cms/HasLock.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Data\Data;
+use Kirby\Toolkit\A;
+use Kirby\Toolkit\F;
+
+trait HasLock
+{
+
+    public function lock($user = false, string $language = null)
```

Comment:
In my opinion, the `lock()` method and *especially* the `unlock()` method do way too much stuff. Also it's completely intransparent which combination of arguments gets or sets stuff. If called incorrectly (e.g. `null` instead of `false`), the methods could break in unexpected ways.

I see you had different get and set methods previously. To be honest I'd return to separate `lock()` and `getLock()` methods (same for unlocking). The setters could accept `string? $user`, which would also accept `null` for "unset".

Reply:
As you saw, I had those and then switched to having only one for each as suggested by @bastianallgeier on Slack. I preferred explicit ones as well, so let’s hear from Bastian.